File: backend/src/main/kotlin/com/example/backend/dtos/VenteDto.kt
package com.example.backend.dtos

import org.bouncycastle.util.test.FixedSecureRandom.BigInteger
import java.math.BigDecimal
import java.time.LocalDate
import java.time.LocalDateTime

data class ProduitAssocieDto(
  val produitId: Long?,
  val quantite: Int?,
  val rayonId: String?,
  val type: String?,
  val prixUnit: Int?,
  val reduction: Int? // Modifiable unit price
)

data class VenteRequestDto(
  val reduction: Int?,
  val clientInfo: ClientInfo,
  val prescripteurInfo: PrescripteurInfo,
  val reductionEnabled: Boolean,
  val prixTotal: Double,
  val prixReduction: Double,
  val commentaire: String,
  val etat: String, // State of the sale: "COMPTANT", "ASSURANCE", or "CREDIT"
  val produits: List<ProduitAssocieDto> // List of associated products with quantity and price
)

data class ClientInfo(
  val id: Int?,
  val type: String?,
  val name: String?,
  val phone: String?,
)

data class PrescripteurInfo(
  val id: Int?,
  val type: String?,
  val name: String?,
)

data class EncaissementRequestDto(
    val typePaiement: String,
    val montantPercu: Int,
    val reste: Int,
    val montantTtc: Int,
    val espece: Int?, // Cash payment amount
    val electronique: ElectroniqueRequestDto?, // Electronic payment details
    val ticket: Int? // Ticket payment amount
)

data class ElectroniqueRequestDto(
    val numeroTelephone: String,
    val montant: Int
)

data class EncaissementDto(
  var typeEncaissement: String,
  val venteId: Long,
  val montantPercu: Int,
  val montantRendu: Int,

  val espece: Int?, // Cash payment amount
  val electronique: ElectroniqueDto?, // Electronic payment details
  val ticket: TicketDto? // Ticket payment amount
)

data class EncaissementDirectDto(
  var encaissementDto: EncaissementDto,
  val venteRequestDto: VenteRequestDto,
)

data class ElectroniqueDto(
  val numeroTelephone: String,
  val montantElectronique: Int
)

data class TicketDto(
  val numeroTicket: String,
  val montantTicket: Int
)

File: backend_py/pharmaPython/app/schemas/vente_dto.py
from pydantic import BaseModel
from typing import Optional, List, Dict, Any

class ProduitAssocieDto(BaseModel):
  produitId: Optional[int] = None
  quantite: Optional[int] = None
  rayonId: Optional[str] = None
  type: Optional[str] = None
  prixUnit: Optional[int] = None
  reduction: Optional[int] = None

class ClientInfo(BaseModel):
  id: Optional[int] = None
  type: Optional[str] = None
  name: Optional[str] = None
  phone: Optional[str] = None

class PrescripteurInfo(BaseModel):
  id: Optional[int] = None
  type: Optional[str] = None
  name: Optional[str] = None

class ElectroniqueRequestDto(BaseModel):
  numeroTelephone: str
  montant: int

class ElectroniqueDto(BaseModel):
  numeroTelephone: str
  montantElectronique: int

class TicketDto(BaseModel):
  numeroTicket: str
  montantTicket: int

class EncaissementRequestDto(BaseModel):
  typePaiement: str
  montantPercu: int
  reste: int
  montantTtc: int
  espece: Optional[int] = None
  electronique: Optional[ElectroniqueRequestDto] = None
  ticket: Optional[int] = None

class EncaissementDto(BaseModel):
  typeEncaissement: str
  venteId: int
  montantPercu: int
  montantRendu: int
  espece: Optional[int] = None
  electronique: Optional[ElectroniqueDto] = None
  ticket: Optional[TicketDto] = None

class VenteRequestDto(BaseModel):
  reduction: Optional[int] = None
  clientInfo: ClientInfo
  prescripteurInfo: PrescripteurInfo
  reductionEnabled: bool
  prixTotal: float
  prixReduction: float
  commentaire: str
  etat: str
  produits: List[ProduitAssocieDto]

class EncaissementDirectDto(BaseModel):
  encaissementDto: EncaissementDto
  venteRequestDto: VenteRequestDto

class VentePageableCustomlDto(BaseModel):
  # Page<Map<String, Any?>> → liste de dicts + méta
  content: List[Dict[str, Any]] = []
  totalElements: int
  totalPages: int
  pageSize: int
  pageNumber: int
  totalAmount: float
  data: Dict[str, Any]
